cmd/minishift/cmd/image/prune.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright (C) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/cmd/minishift/state"
+	"github.com/minishift/minishift/pkg/minikube/cluster"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/docker/image"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	imagePruneCmd = &cobra.Command{
+		Use:   "prune [image ...]",
+		Short: "Prunes the specified container images.",
+		Long:  "Prunes the specified container images.",
+		Run:   pruneImage,
+	}
+)
+
+const (
+	noImageProvided = "You need to either specify a list of images or a single image on the command line"
+)
+
+func pruneImage(cmd *cobra.Command, args []string) {
+	api := libmachine.NewClient(state.InstanceDirs.Home, state.InstanceDirs.Certs)
+	defer api.Close()
+
+	util.ExitIfUndefined(api, constants.MachineName)
+
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating the VM client: %v", err))
+	}
+
+	util.ExitIfNotRunning(host.Driver, constants.MachineName)
+
+	envMap, err := cluster.GetHostDockerEnv(api)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error determining Docker daemon settings: %v", err))
+	}
+
+	var out io.Writer
+	if logToFile {
+		logFile := createLogFile()
+		defer logFile.Close()
+		out = logFile
+	} else {
+		out = os.Stdout
+	}
+
+	images := imagesToPrune(api, args)
+
+	handler, err := image.NewOciImageHandler(host.Driver, envMap)
```

Comment:
This does not look right to me. We should not need docker daemon to delete a cached docker image. Need to look in to it more.

Reply:
@praveenkumar Have you used `NewLocalOnlyOciImageHandler`?

```
// NewLocalOnlyOciImageHandler creates  a new ImageHandler which can only interact with the local cache.
// No connection information to the Docker daemon are provided. Functions interacting with the Docker daemon will fail.
func NewLocalOnlyOciImageHandler() (*OciImageHandler, error) {
	return &OciImageHandler{driver: nil, dockerClientSettings: nil}, nil
}
```

Reply:
@LalatenduMohanty Check https://github.com/minishift/minishift/pull/2381#discussion_r187993721